api/v3/utils.php

```diff
@@ -2078,9 +2078,14 @@ function _civicrm_api3_validate_integer(&$params, $fieldName, &$fieldInfo, $enti
         $fieldValue = NULL;
       }
     }
-    if (!empty($fieldInfo['pseudoconstant']) || !empty($fieldInfo['options']) || $fieldName === 'campaign_id') {
+    if (
+    (!empty($fieldInfo['pseudoconstant']) || !empty($fieldInfo['options']) || $fieldName === 'campaign_id')
+     // if it is already numeric AND it is an FK field we don't need to validate because
+     // sql will do that for us on insert (this also saves a big lookup)
+     && (!is_numeric($fieldValue) || empty($fieldInfo['FKClassName']))
```

Comment:
I have re-written this line 8 times so I'm gonna write out what I'm doing

We will validate if it's a a pseudoconstant, has options or is campaign id 

AND it is not an FK OR the value is not numeric

Ie if it is an FK & we have a string then validate
if it's numeric & it's controlled by option values then validate

but don't validate a numeric FK field